### PR TITLE
You can no longer print infinite ammunition.

### PR DESCRIPTION
**Pull request by DaxDupont**
[Changelogs]: 

:cl: Dax Dupont
balance: After successful lobbying by the Space Ancap Federation ammo is no longer free. Material redemption cost is now determined by the amount of bullets left.
balance: .38 ammo is now cheaper to produce. 
balance: Casings now give a little bit of metal. mostly for immersion sake.
/:cl:

[why]: After seeing ye olde print infinite ammo murderboning I figured this would be a change that people might get behind. 
Things like the .357 are quite powerful, and being able to print and shoot infinite amounts of ammo by reclaiming the full amount on an empty mag seems obscene.
Might tweak the base cost if someone has input.

### CODE REVIEW

File: code/modules/projectiles/boxes_magazines/_box_magazine.dm [hunk 1]
```diff
@@ -21,9 +21,19 @@
 	var/caliber
 	var/multiload = 1
 	var/start_empty = 0
+	var/list/bullet_cost = list() 
+	var/list/base_cost = list() // override this one as well if you override bullet_cost
```
Comment (vuonojenmustaturska) [inline code/modules/projectiles/boxes_magazines/_box_magazine.dm]: initialize these lists as null, otherwise you're throwing memory away

**Review by vuonojenmustaturska — DISMISSED** [review]

**Comment by BeeSting12** [comment]
Nerf to the detective. Probably healthy for the game but 👎 because i like abusing this

**Comment by DaxDupont** [comment]
@BeeSting12 it could probably be overwritten to be less painful for the detective's ammo. 
If anyone wants to give me numbers that would be acceptable.

**Comment by DaxDupont** [comment]
@vuonojenmustaturska ree now i need to redo parts

**Comment by Repukan** [comment]
Makes no sense that ammo you're supposed to use sensibly or spend more tc refilling it is easily available from normal lathes. If you want to make it a worthwhile nerf, just remove it from hacked lathes.

**Comment by Bombpants** [comment]
Has it really been abused?

**Comment by AlexTheSysop42** [comment]
>Makes no sense that ammo you're supposed to use sensibly or spend more tc refilling it is easily available from normal lathes. If you want to make it a worthwhile nerf, just remove it from hacked lathes.

It's supposed to be balanced around being really expensive to make, maybe have the nerf only apply to .357 loaders because that's the one that's abused

**Comment by AlexTheSysop42** [comment]
>Has it really been abused?

>After seeing ye olde print infinite ammo murderboning I figured this would be a change that people might get behind.

**Comment by DaxDupont** [comment]
@AlexTheSysop42 might make more sense to make detective .38 cheaper instead.

**Comment by ExcessiveUseOfCobblestone** [comment]
:( good change but dang that sucks lol

**Comment by Hatfish** [comment]
this is a good change, detectives should learn to play their roles.  don't act like you're a cop with ballistics if you don't want to run out of ammo, slick.

**Comment by ExcessiveUseOfCobblestone** [comment]
They aren't going to run out of ammo, they're just going to eat up your metal reserves

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by Mickyan** [comment]
Should the autolathe even be able to produce lethal ammo? Now that it's public access in most maps it seems overkill, even if you need to hack it first

**Comment by Spaceinaba** [comment]
lethal ammo does nothing without a gun so I don't see the issue
we have no productive way to craft our own ammo with gunpowder and metal so for the moment I don't see an issue with having it be in lathes

**Comment by ExcessiveUseOfCobblestone** [comment]
@Mickyan ammo doesn't kill people so it's not malicious on it's own anymore than a timer that only goes on grenades and TTVs (maybe circuits?).

**Comment by Iamgoofball** [comment]
I used to do this back in the day, good riddance

**Comment by Anonmare** [comment]
Metal is ludicrously common anyway

**Comment by Kromgar** [comment]
I mean does it matter? One miner can bring back 2,000 metal

**Comment by KorPhaeron** [comment]
Conservation of matter

**Comment by RandomMarine** [comment]
Extra magazines/speedloaders from uplink only.
Make lathes only make the ammo with no container.

**Comment by Kromgar** [comment]
@RandomMarine great compromise

**Comment by improvedname** [comment]
>realistic change not being able to refund a speedloader/ammo box for the metal your orginally used

>autistic reaction nobody wants such change because it makes to difficult for them yet they say metal is easy to get yet they dont want it

>pacifist change remove 357. From autolathe

**Comment by Shadowflame909** [comment]
>can make infinite ammo without any cost

Who put them into the autolathe without accounting for this??

**Comment by ExcessiveUseOfCobblestone** [comment]
Its not actually 0 cost, you can just get a full refund if you keep the loader

**Comment by Shadowflame909** [comment]
Well thats pretty silly, but I say this PR seems pretty much needed. Everything in ss13 has to have some sort of cost!

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**